lib/client/connection.js

```diff
@@ -584,3 +597,44 @@ Connection.prototype._firstQuery = function(fn) {
     }
   }
 };
+
+Connection.prototype._firstSnapshotRequest = function (fn) {
+  for (var id in this.snapshotRequests) {
+    var snapshotRequest = this.snapshotRequests[id];
+    if (fn(snapshotRequest)) {
+      return snapshotRequest;
+    }
+  }
+};
+
+/**
+ * Get a read-only snapshot at a given version or time
+ *
+ * @param collection - the collection name of the snapshot
+ * @param id - the ID of the snapshot
+ * @param version - the version number, or Date of the snapshot to fetch. If an exact version or Date match is not made,
+ *   then the next lowest version is returned. ie if a document has 6 versions, asking for v7 will return v6. If ops
+ *   were submitted at 02:00 and 03:00, then asking for a Date at 02:30 will return the 02:00 version.
+ * @param callback - (error, snapshot) => void, where snapshot takes the following schema:
+ *
+ * {
+ *   collection: string; // collection name of the snapshot
+ *   id: string;         // ID of the snapshot
+ *   version: number;    // version number of the snapshot
+ *   timestamp: number;  // the UNIX timestamp of the snapshot
+ *   deleted: boolean;   // true if the returned version is a deleted snapshot
+ *   data: any;          // the snapshot
+ * }
+ *
+ */
+Connection.prototype.getSnapshot = function(collection, id, version, callback) {
```

Comment:
Could `version` be an optional param?

Reply:
It looks like it is an optional param, with test coverage for the `undefined` case: https://github.com/share/sharedb/pull/220/files#diff-9ca1c4bd7a12ccd08af041e04970dfe7R146